lintreview/docker.py

```diff
@@ -44,45 +58,50 @@ def apply_base(value):
 
 
 def image_exists(name):
-    """Check if a docker image exists"""
-    process = subprocess.Popen(
-        ['docker', 'images', '-q', name],
-        stdin=subprocess.PIPE,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        shell=False)
-    output, error = process.communicate()
-    return len(output) > 0
+    # type: (str) -> bool
+    """Check if a docker image exists."""
+    client = _get_client()
+    try:
+        client.images.get(name)
+    except ImageNotFound:
+        return False
+    return True
 
 
 def images():
-    """Get the docker image list"""
-    process = subprocess.Popen(
-        ['docker', 'images'],
-        stdin=subprocess.PIPE,
-        stdout=subprocess.PIPE,
-        stderr=subprocess.PIPE,
-        shell=False)
-    output, error = process.communicate()
-    return output.decode('utf8')
+    # type: () -> List[str]
+    """Get the docker image list."""
+    client = _get_client()
+    d_images = client.images.list()
```

Comment:
What does the d prefix mean?

Reply:
docker, I'm just trying to avoid shadowing `images` which is the name of this function.

Reply:
I cleaned up that function a bit to avoid the assignment and just use it in the for loop, should read much clearer now.